client/components/payment-methods-list/payment-method.tsx

```diff
@@ -23,6 +23,70 @@ import Pill from '../pill';
 import PaymentMethodDisabledTooltip from '../payment-method-disabled-tooltip';
 import './payment-method.scss';
 
+function PaymentMethodLabel( {
+	label,
+	required,
+	status,
+	disabled,
+	id,
+}: {
+	label: string;
+	required: boolean;
+	status: string;
+	disabled: boolean;
+	id: string;
+} ): React.ReactElement {
```

Comment:
Good effort to extract this part to a separate function.
Nitpick: may use `const`here. 

```suggestion
const PaymentMethodLabel = ( {
	label,
	required,
	status,
	disabled,
	id,
}: {
	label: string;
	required: boolean;
	status: string;
	disabled: boolean;
	id: string;
} ): React.ReactElement => {
```